sdk/provisioning/Azure.Provisioning/src/Generated/ResourceGroup.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable enable
+
+using Azure.Core;
+using Azure.Provisioning;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Primitives;
+using System;
+using System.ComponentModel;
+
+namespace Azure.Provisioning.Resources;
+
+/// <summary>
+/// ResourceGroup.
+/// </summary>
+public partial class ResourceGroup : Resource
+{
+    /// <summary>
+    /// The name of the resource group to create or update. Can include
+    /// alphanumeric, underscore, parentheses, hyphen, period (except at end),
+    /// and Unicode characters that match the allowed characters.
+    /// </summary>
+    public BicepValue<string> Name { get => _name; set => _name.Assign(value); }
+    private readonly BicepValue<string> _name;
```

Comment:
One of the really important parts of `BicepValue` is that we don't exchange references.  You can see here we do an `.Assign` rather than `=`.  This lets us track things like where the current value comes from and where the value it was just assigned came from.  This is already crucial to correctly resolving when something should be a reference vs. literal value, but it'll also be really important to understanding when someone assigns a value that crosses module boundaries and we need to implicitly inject an extra output/param.